app/controllers/pets_controller.rb

```diff
@@ -11,8 +11,13 @@ def show
 
   def create
     shelter = Shelter.find(params[:shelter_id])
-    shelter.pets.create(pet_params)
-    redirect_to "/shelters/#{shelter.id}/pets"
+    pet = shelter.pets.create(pet_params)
+    if pet.valid?
```

Comment:
Adds a conditional and flash messages to indicate which fields were missing when submitting the form